pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -84,14 +85,122 @@ func NewOperatorStepResolver(lister operatorlister.OperatorLister, client versio
 	return stepResolver
 }
 
-func (r *OperatorStepResolver) ResolveSteps(namespace string) ([]*v1alpha1.Step, []v1alpha1.BundleLookup, []*v1alpha1.Subscription, error) {
+type walkOption func(csv *v1alpha1.ClusterServiceVersion) error
+
+func WithCSVPhase(phase v1alpha1.ClusterServiceVersionPhase) walkOption {
+	return func(csv *v1alpha1.ClusterServiceVersion) error {
+		if csv == nil || csv.Status.Phase != phase {
+			return fmt.Errorf("clusterServiceVersion %s/%s in phase %s instead of %s", csv.GetNamespace(), csv.GetName(), csv.Status.Phase, phase)
```

Comment:
nit: `ClusterServiceVersion` to have consistent capitalization? or maybe simply `csv`?

Reply:
I'll change it to `csv`.